app/src/org/commcare/adapters/IncompleteFormListAdapter.java

```diff
@@ -168,15 +168,28 @@ public void resetRecords() {
 
         // Sort FormRecords by modification time, most recent first.
         Collections.sort(records, (left, right) -> {
-            long leftModTime = left.lastModified().getTime();
-            long rightModTime = right.lastModified().getTime();
-
-            if (leftModTime > rightModTime) {
-                return -1;
-            } else if (leftModTime == rightModTime) {
-                return 0;
+            if (!filter.equals(FormRecordFilter.Pending)) {
+                long leftModTime = left.lastModified().getTime();
+                long rightModTime = right.lastModified().getTime();
+
+                if (leftModTime > rightModTime) {
+                    return -1;
+                } else if (leftModTime == rightModTime) {
+                    return 0;
+                } else {
+                    return 1;
+                }
             } else {
-                return 1;
+                int leftSubmissionOrderNum = left.getSubmissionOrderingNumber();
```

Comment:
We also do this [sorting](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/utils/StorageUtils.java#L95) when sending the forms to the server. It will be great if we can utilize the same code here.

Reply:
Used sorting from StorageUtils and also added to StorageUtils function for sorting by last modified time.